### PR TITLE
Add empty_value parameter to CSV processor

### PR DESCRIPTION
**Pull request by probakowski**
This change adds `empty_value` parameter to the CSV processor.
This value is used to fill empty fields. Fields will be skipped
if this parameter is omitted. This behavior is the same for both
quoted and unquoted fields.

### CODE REVIEW

File: modules/ingest-common/src/test/java/org/elasticsearch/ingest/common/CsvProcessorTests.java [hunk 1]
```diff
@@ -158,7 +218,7 @@ public void testEmptyFields() throws Exception {
         items.keySet().stream().skip(numItems - 1).forEach(key -> assertFalse(ingestDocument.hasField(key)));
     }
 
-    public void testWrongStings() throws Exception {
+    public void testWrongStings() {
```
Comment (droberts195) [inline modules/ingest-common/src/test/java/org/elasticsearch/ingest/common/CsvProcessorTests.java]: typo: `Sting` -> `String`

(But it was there before and you've got a green build so if no other changes are required then no need to trigger a rebuild just for this.)

**Review by droberts195 — APPROVED** [review]
LGTM

I also tested it in conjunction with elastic/elasticsearch#51492 and elastic/kibana#56038 and confirmed that the end-to-end import process now works with CSV files that have quoted empty fields and also those that have unquoted empty fields. 🎉

File: docs/reference/ingest/processors/csv.asciidoc [hunk 2]
```diff
@@ -13,6 +13,7 @@ Extracts fields from CSV line out of a single text field within a document. Any
 | `quote`           | no        | "        | Quote used in CSV, has to be single character string
 | `ignore_missing`  | no        | `true`   | If `true` and `field` does not exist, the processor quietly exits without modifying the document
 | `trim`            | no        | `false`  | Trim whitespaces in unquoted fields
+| `empty_value`     | no        | -        | Value used to fill empty fields, empty fields will be skipped if this is not provided
```
Comment (jbaiera) [inline docs/reference/ingest/processors/csv.asciidoc]: Could we add a quick mention here what we consider empty fields? ("no value" and "empty quotes"). This way we have it spelled out which fields are officially considered empty.

**Review by jbaiera — APPROVED** [review]
One quick suggestion for the docs, but otherwise LGTM

**Comment by probakowski** [comment]
@elasticmachine update branch

File: docs/reference/ingest/processors/csv.asciidoc (hunk 2 above)
Comment (probakowski) [reply docs/reference/ingest/processors/csv.asciidoc]: Good idea, I've added it

**Comment by probakowski** [comment]
@elasticmachine update branch

**Pull request merged**